_rules/active-focus-shortcut-1e9941.md

```diff
@@ -0,0 +1,237 @@
+---
+id: 1e9941
+name: Printable key shortcut is only active when component has focus
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts for a user interface component are implemented using only printable characters, then they are only available when that component has focus.
+accessibility_requirements: # Remove whatever is not applicable
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+  - CSS Styling
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [HTML document][] with at least one [keyboard shortcut][] that requires pressing only [printable character][] keys.
+
+## Expectation
+
+For each [user interface component][] that is a [descendent][] of the root node of the test target, if a [printable character][] [shortcut][keyboard shortcut] exists to trigger the action of that [user interface component][], it can only be triggered when that component has [focus][].
+
+## Assumptions
+
+This rule assumes as applicable [keyboard shortcuts][keyboard shortcut] those implemented by the test target [content][]. Any other means (e.g. browser extensions, browser settings, user agents, external browser applications) are not considered.
+
+## Accessibility Support
+
+_There are no major accessibility support issues known for this rule._
+
+## Background
+
+- [Understanding Success Criterion 2.1.4: Character Key Shortcuts](https://www.w3.org/WAI/WCAG21/Understanding/character-key-shortcuts.html)
+- [G127 Providing a mechanism to allow users to remap or turn off character key shortcuts](https://www.w3.org/WAI/WCAG21/Techniques/general/G217)
+
+## Test Cases
+
+### Passed
+
+#### Passed Example 1
+
+The [HTML document][] has a single [printable character][] [keyboard shortcut][] for a [user interface component][], which is only available when that component has focus.
+
+```html
+<html>
+  <head>
+    <title>Passed Example 1 for rule 1e9941</title>
+    <script>
+      function shortcut(params) {
+        document.body.addEventListener("keydown", function(event) {
+          const target = document.getElementById("target");
+
+          if (
+            event.key === params.shortcutKey &&
+            (!!!params.ctrlKey || event.ctrlKey) &&
+            (!!!params.focusOnly || document.activeElement === target)
+          ) {
+            document.getElementById("list").innerHTML +=
+              "<li>" + target.value + "</li>";
+            target.value = "";
+            event.preventDefault();
+          }
+        });
+      }
+    </script>

```

Comment:
Now that this code is used in most examples, in all of the rules (I think), it can be moved to a separate test-assets file. That will DRY the code, shorten the rules, and make the repetition obvious (hence easier to read).

I think that the `onload` attribute in the `body` has a neat enough syntax that there is no need to read the JS every single time.